open-craft/sprints
# [BB-3173] Fix calculating vacations and commitments

## What it is
This contains the following changes:
1. Fixes calculating commitments and vacations with the assumption that a new sprint starts at midnight UTC.
1. Upgrades Python to `3.9` and Django to `3.1.3` and bumps dependencies (minor changes; unfortunately we cannot use [PEP 585](https://www.python.org/dev/peps/pep-0585/), because mypy [does not support it yet](https://github.com/python/mypy/issues/7907)).
1. Removes unused code from the cookiecutter template.

## Testing instructions 
1. Set `.env` and `frontend/env.local` (you can find them in a comment on the ticket).
1. Comment out [these two lines](https://github.com/open-craft/sprints/blob/master/frontend/src/App.js#L87-L88). They're irrelevant to the scope and will only slow down the local environment.
1. Start the backend with `docker-compose -f local.yml up --build`.
1. Register new account and get the verification link with `docker-compose -f local.yml logs django`.
1. Run migrations with `docker-compose -f local.yml exec django python manage.py migrate`.
1. Navigate to `frontend` directory and run `npm i && npm start`.
1. Log in and navigate to cell's board.
1. Create an "off" calendar event for 4 edge cases described in the `_get_vacation_for_day`'s docstring. Change your timezone [here](https://docs.google.com/spreadsheets/d/107dR9H1vWjLpJlXPuBaFJIFaEPEmaSh50xLow_aEGVw/edit#gid=1116869903) to a negative one to see how it impacts these cases.
